curriculum/challenges/english/02-javascript-algorithms-and-data-structures/es6/create-a-javascript-promise.english.md

```diff
@@ -0,0 +1,56 @@
+---
+id: 5cdafbb0291309899753167f
+title: Create a JavaScript Promise
+challengeType: 1
+---
+
+## Description
+<section id='description'>
+A promise in JavaScript is exactly what it sounds like. You use it to make a promise to do something, usually asynchronously. When the task completes you either fulfill your promise or fail to do so. <code>Promise</code> is a constructor function so you need to use the <code>new</code> keyword to create one. It takes a function as its argument with two parameters, <code>resolve</code> and <code>reject</code>. These are methods used to determine the outcome of the promise. The syntax looks like this:
+
+```js
+const myPromise = new Promise((resolve, reject) => {
+
+});
+```
+
+</section>
+
+## Instructions
+<section id='instructions'>
+Create a new promise called <code>makeServerRequest</code>. Pass in a function with <code>resolve</code> and <code>reject</code> parameters to the constructor.
+</section>
+
+## Tests
+<section id='tests'>
+
+```yml
+tests:
+  - text: Your promise should be in a variable called <code>makeServerRequest</code>.
```

Comment:
I notice if you leave off the variable declaration (i.e. `var`, `const`, `let`) the first test fails, but since the "variable not defined" error in the browser console does not show up (as it really should), it might be a little confusing to what the actually problem is.  I think the following change in the test will make it clearer.
```suggestion
  - text: You should assign a promise to a declared variable named <code>makeServerRequest</code>.
```